frontend/src/components/program/DialogActivityEdit.vue

```diff
@@ -106,13 +137,45 @@ export default {
           period: entry.period()._meta.self,
           start: entry.start,
           end: entry.end,
-          activity: this.activity._meta.self,
+          activity: this.currentActivity._meta.self,
         })
       })
 
+      if (!this.hideHeaderFields) {
+        const untouchedActivityResponsibles = intersectionWith(
+          this.initialResponsibles,
+          this.entityData.activityResponsibles,
+          (a, b) => {
+            return a.campCollaboration._meta.self === b.campCollaboration._meta.self
+          }
+        )
+
+        const differentActivityResponsibles = differenceWith(
+          [...this.initialResponsibles, ...this.entityData.activityResponsibles],
+          untouchedActivityResponsibles,
+          (a, b) => {
+            return a.campCollaboration._meta.self === b.campCollaboration._meta.self
+          }
+        )
+
+        differentActivityResponsibles.forEach((activityResponsible) => {
+          if (activityResponsible?._meta?.self) {
+            promises.push(this.api.del(activityResponsible._meta.self))
+          } else {
+            promises.push(
+              this.activityResponsibles.$post({
+                activity: this.currentActivity._meta.self,
+                campCollaboration: activityResponsible.campCollaboration._meta.self,
+              })
+            )
+          }
+        })
+      }
```

Comment:
I think this is solved little to simple. (even though i see a lot of thought process)
1. This should not be in the update logic of the scheduleEntries
2. In this function all the promises to update things (scheduleEntries, the activity itself) are collected and awaited. This should be preserved.

I would split the 2 changes in separate PR and first add the possibility to edit the scheduleEntries in the Activity View.
And then add the responsibles to the activty dialog in the picasso.

Reply:
@BacLuc "too simple"? What should I make more complicated/different?

> 1. This should not be in the update logic of the scheduleEntries

Why?

> 2. … This should be preserved.

It is, it just adds the update scheduleEntries to the same promises array as scheduleEntries and activity.

Reply:
> @BacLuc "too simple"? What should I make more complicated/different?
> 
> > 1. This should not be in the update logic of the scheduleEntries
> 
> Why?

But you don't want to send the post/delete requests for the activityResponsibles for each ScheduleEntry.
(The delete and posts for the same insertion/deletion are sent for each scheduleEntry once as i read the code).


> 
> > 2. … This should be preserved.
> 
> It is, it just adds the update scheduleEntries to the same promises array as scheduleEntries and activity.

but something seems to not work:
[Screencast from 07.01.2024 14:24:44.webm](https://github.com/ecamp/ecamp3/assets/1506818/93fa24e1-5dc8-4736-9f3d-bfe399bfbb59)

Reply:
uups, sorry.
Its not in the update for the scheduleentries.